### PR TITLE
feat(Forms): auto-open iterate container items when validation errors and make `Iterate.Toolbar` fully customizable

### PR DESCRIPTION
**Pull request by tujoworker**
- This PR allows a dev to fully customize the buttons for the edit and view container toolbar.
- I also auto-opens the edit container, when there is a validation error.
- A followup-up PR #3919 has been created.

Quick example of a fully customized Toolbar:

```tsx
<Form.Handler>
  <Iterate.Array path="/countries" defaultValue={[null]}>
    <Iterate.ViewContainer
      toolbar={
        <Iterate.Toolbar>
          {({ items }) => {
            if (items.length === 1) {
              return <Iterate.ViewContainer.EditButton />
            }
            return (
              <>
                <Iterate.ViewContainer.EditButton />
                <Iterate.ViewContainer.RemoveButton />
              </>
            )
          }}
        </Iterate.Toolbar>
      }
    >
      <Value.SelectCountry
        label="Land du er statsborger i"
        itemPath="/"
      />
    </Iterate.ViewContainer>

    <Iterate.EditContainer
      toolbar={
        <Iterate.Toolbar>
          {({ items }) => {
            if (items.length === 1) {
              return null
            }

            return (
              <>
                <Iterate.EditContainer.DoneButton />
                <Iterate.EditContainer.CancelButton />
              </>
            )
          }}
        </Iterate.Toolbar>
      }
    >
      <Field.SelectCountry
        label="Land du er statsborger i"
        itemPath="/"
        required
      />
    </Iterate.EditContainer>
  </Iterate.Array>

  <Iterate.PushButton
    path="/countries"
    pushValue={null}
    text="Legg til flere statsborgerskap"
  />
</Form.Handler>
```

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx [hunk 1]
```diff
@@ -58,3 +58,9 @@ With an optional `title` and [Iterate.Toolbar](/uilib/extensions/forms/Iterate/T
 ### Value composition
 
 <Examples.ValueComposition />
+
+### With initial value and `containerMode="new"`
+
+By using `containerMode="new"` and providing an initial value of `null` to the array, th user needs to fill out the first item.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx:64]:
```suggestion
By using `containerMode="new"` and providing an initial value of `null` to the array, the user needs to fill out the first item.
```

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Array/types.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Path, UseFieldProps } from '../../types'
 import { Props as FlexContainerProps } from '../../../../components/flex/Container'
 
-export type ContainerMode = 'view' | 'edit'
+export type ContainerMode = 'view' | 'edit' | 'new'
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/Array/types.ts:4]: I see that this type is used many places, is the new value '`new'` supported in all of these places?
If so, then there's a few places we can/should update the docs, do a search in vscode(command+shift+f) for "Can be `view` or `edit`". We should probably add `'new'` there as well.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx [hunk 3]
```diff
@@ -364,3 +364,105 @@ export const WithVisibility = () => {
...
+            </Iterate.EditContainer>
+          )
+        }
+
+        const CreateNewEntry = () => {
+          return (
+            <Iterate.PushContainer
+              path="/countries"
+              openButton={
+                <Iterate.PushContainer.OpenButton text="Legg til flere statsborskap" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx:397]:
```suggestion
                <Iterate.PushContainer.OpenButton text="Legg til flere statsborgerskap" />
```

**Review by langz — APPROVED** [review]
When demoing the [example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#with-initial-value-and-containermodenew), I've experienced a few things:

If I have two fields without values, only one is marked in red/error(could perhaps be logical, as maybe only the 1st one is required):

https://github.com/user-attachments/assets/86502384-10b8-4671-8602-cc36ca1877b0

The input field behaves quite differently if I've have clicked send inn or not.
When not clicking "send inn", the input field behaves as expected(not showing "ferdig" and "avbryt" button). But when first pressing "send inn"-button I get presented with the "ferdig" and "avbryt"-button, and further I'm able to remove the data/field, and when pressing "send inn"-button again, I get no error:

https://github.com/user-attachments/assets/5d1ac33a-5cca-46a8-87da-70e5abee75bd




These are just a few experiences/observations, not saying it's anything critical or "errors".

**Comment by tujoworker** [comment]
Thank you for the good feedback. With the changes made, it looks good to me. Would love it you gave it another go. Here is [the example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#with-initial-value-and-containermodeinitial-open).

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Array/types.ts (hunk 2 above)
Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/Iterate/Array/types.ts]: - `new` was renamed to `initial-open`.

I think we are good there documentation wise. The `Form.Section` has also a `ContainerMode`, which does not support `initial-open` (at the moment), but there is this different prop `validateFieldsInitially`, which is kind of similar ([PR](https://github.com/dnbexperience/eufemia/pull/3878)), as it keeps the container "open", when there are errors. Maybe we actually should replace `validateFieldsInitially` in favor of something like `containerMode="validate-fields-initially"` 🤔

Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/Iterate/Array/types.ts]: Or even: `containerMode="openWhenFieldsError"`

**Comment by langz** [comment]
> Thank you for the good feedback. With the changes made, it looks good to me. Would love if you gave it another go. Here is [the example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open).

Thanks, @tujoworker 🥇 
Looks great, what I reported works perfectly now 👏 

I do however experience that [the example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open) laggs or is "stuttering".
It seems like it re-renders or so? 🤔 

https://github.com/user-attachments/assets/b1cf2f17-3da9-4187-ab93-7f8d4d57d821

**Comment by tujoworker** [comment]
> I do however experience that [the example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open) laggs or is "stuttering". It seems like it re-renders or so?

This should now be solved as well. Can you have another look? 🙏

**Comment by tujoworker** [comment]
We ended up do always witch to edit mode if there is a validation error in the edit container. We also added three new properties in overall:

- `minimumRequiredItems={1}`
- `hideToolbarWhen={(items) => items.length === 1}` 
- `containerMode` mroe for internal usage

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx [hunk 4]
```diff
@@ -88,6 +88,16 @@ render(
 )
 ```
 
+### Initial container mode
+
+By default, the container mode is set to `auto`. This means that the container will open (switch to `edit` mode) when there is an error in the container or the value is falsy (empty string, null, undefined, etc.).
+
+When a new item is added via the [Iterate.PushButton](/uilib/extensions/forms/Iterate/PushButton/) component, the item before it will change to `view` mode, if it had no validation errors.
+
+You can also set `minimumRequiredItems` to 1, to ensure that the first has not remove button as long as there is at least one item.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx:97]:
```suggestion
You can also set `minimumRequiredItems` to 1, to ensure that the first item has not remove button as long as there is at least one item.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx [hunk 5]
```diff
@@ -88,6 +88,16 @@ render(
...
+
+You can also set `hideToolbarWhen` to `(index, items) => items.length === 1` to hide the toolbar when there is only one item.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx:99]:
```suggestion
You can also set `hideToolbarWhen`, like `(index, items) => items.length === 1` to hide the toolbar when there is only one item.
```

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Array/__tests__/__image_snapshots__/iteratearray-have-to-match-view-container.snap.png
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/Array/__tests__/__image_snapshots__/iteratearray-have-to-match-view-container.snap.png:1]: Is this change made on purpose?
I haven't looked into it, but the screenshot looks quite different now than how it looked before this change.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx (hunk 4 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx:97]: Is item the correct word here? 🤔

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Array/ArrayDocs.ts [hunk 6]
```diff
@@ -41,6 +41,21 @@ export const ArrayProperties: PropertiesTableProps = {
     type: 'unknown',
     status: 'optional',
   },
+  minimumRequiredItems: {
```
[the remark this answers is not in the document]
Comment (langz) [reply packages/dnb-eufemia/src/extensions/forms/Iterate/Array/ArrayDocs.ts:44]: I tried using the new props on the example above, but nothing really happened/changed:

image

Is it only in certain/special scenarios where I can use these new props?

[the remark this answers is not in the document]
Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/Iterate/Array/ArrayDocs.ts:44]: Yes, only on containers.

**Comment by langz** [comment]
I tried linking/visiting a specific heading in the docs, like Primitive items as fields like so:
https://eufemia.dnb.no/uilib/extensions/forms/Iterate/Array/demos/#primitive-items-as-fields

But when visiting the same link in the deploy preview:
https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#primitive-items-as-fields

It doesn't really work, I get scrolled to a different example.

Just consider if this is something worth fixing/improving. If not, then ignore my comment 🙏

**Comment by langz** [comment]
When comparing the ["Toggle between a view and edit container" at "main"](https://eufemia.dnb.no/uilib/extensions/forms/Iterate/Array/demos/#toggle-between-a-view-and-edit-container) and in [this deploy preview](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#toggle-between-a-view-and-edit-container):
- They look different(perhaps intended)
- The new example has some kind of animation/flicker between view and edit mode.

https://github.com/user-attachments/assets/a8bd5243-7eca-493a-b672-6646f0fc9f0f

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Array/ArrayDocs.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (langz) [reply packages/dnb-eufemia/src/extensions/forms/Iterate/Array/ArrayDocs.ts:44]: I see, for the following container?
- EditContainer
- PushCOntainer
- ViewContainer

And not for the following containers?
Screenshot 2024-09-09 at 15 46 04

**Comment by langz** [comment]
> > I do however experience that [the example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open) laggs or is "stuttering". It seems like it re-renders or so?
> 
> This should now be solved as well. Can you have another look? 🙏

I was able to experience the same today, when scrolling down to the following [example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open) "Initially open" and then refreshing using command + r (using Chrome):


https://github.com/user-attachments/assets/629fc20c-ec7a-4d2b-b64a-84603b37e9d3

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx [hunk 7]
```diff
@@ -47,6 +47,15 @@ With an optional `title` and [Iterate.Toolbar](/uilib/extensions/forms/Iterate/T
 
 <Examples.ViewAndEditContainer />
 
+### Initially open
+
+This example uses additioanl container props (for the `EditContainer` and the `ViewContainer`) to customize the behavior of the nested containers.
+
+- `minimumContainerItems` will hide the toolbar when the array value (items) has less than this number of items.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx:54]: Will `minimumContainerItems` hide the toolbar or the remove button?
In the following line in the docs, it says "remove button", or maybe it's the same thing?
https://github.com/dnbexperience/eufemia/pull/3877/files#diff-1307180c1bc7887ebd200f1745832774f211c66109aed73d5918609e920d992dR99

Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx:54]: I also don't feel like this description matches the description in the properties docs, 
https://github.com/dnbexperience/eufemia/pull/3877/files#diff-4d6ee7da938da62a990954613151bbe1f58422c8493b7bc65288e11db0e8c78fR44

**Comment by langz** [comment]
> > > I do however experience that [the example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open) laggs or is "stuttering". It seems like it re-renders or so?
> > 
> > 
> > This should now be solved as well. Can you have another look? 🙏
> 
> I was able to experience the same today, when scrolling down to the following [example](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open) "Initially open" and then refreshing using command + r (using Chrome):
> 
>  Screen.Recording.2024-09-09.at.15.52.32.mov

This looks great now, thanks for looking into this ❤️

**Comment by langz** [comment]
> When comparing the ["Toggle between a view and edit container" at "main"](https://eufemia.dnb.no/uilib/extensions/forms/Iterate/Array/demos/#toggle-between-a-view-and-edit-container) and in [this deploy preview](https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#toggle-between-a-view-and-edit-container):
> 
> * They look different(perhaps intended)
> * The new example has some kind of animation/flicker between view and edit mode.
> 
>  Screen.Recording.2024-09-09.at.15.38.21.mov

This looks great now :) 
There's only some slight differences in spacing(less spacing above the "add another account"-button), but that could be intentional 🙏 

Screenshot 2024-09-10 at 09 41 42

**Comment by langz** [comment]
> I tried linking/visiting a specific heading in the docs, like Primitive items as fields like so: https://eufemia.dnb.no/uilib/extensions/forms/Iterate/Array/demos/#primitive-items-as-fields
> 
> But when visiting the same link in the deploy preview: https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#primitive-items-as-fields
> 
> It doesn't really work, I get scrolled to a different example.
> 
> Just consider if this is something worth fixing/improving. If not, then ignore my comment 🙏

It works now, weeee 🥳 🎆

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx (hunk 7 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx]: Yes, very true. I updated the docs 🙏

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Toolbar/ToolbarDocs.ts [hunk 8]
```diff
@@ -1,5 +1,9 @@
 import { PropertiesTableProps } from '../../../../shared/types'
 
-export const ToolbarProperties: PropertiesTableProps = {}
-
-export const ToolbarEvents: PropertiesTableProps = {}
+export const ToolbarProperties: PropertiesTableProps = {
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/Toolbar/ToolbarDocs.ts:3]: We need to add a tab/page for this, I'm not able to see the properties tabs here:
https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Toolbar/demos/

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Toolbar/ToolbarDocs.ts [hunk 9]
```diff
@@ -1,5 +1,9 @@
...
+  variant: {
+    doc: 'Use variants to render the toolbar differently. See the info section for more info.',
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/Toolbar/ToolbarDocs.ts:5]:
```suggestion
    doc: 'Use variants to render the toolbar differently. See the info section for more info.',
```

Which values of variant is available?

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx [hunk 10]
```diff
@@ -22,4 +24,74 @@ render(
 )
 ```
 
-The Toolbar is integrated into the containers [Iterate.ViewContainer](/uilib/extensions/forms/Iterate/ViewContainer/) and [Iterate.EditContainer](/uilib/extensions/forms/Iterate/EditContainer/).
+## Cusomize the Toolbar
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx:27]:
```suggestion
## Customize the Toolbar
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx [hunk 11]
```diff
@@ -22,4 +24,74 @@ render(
...
+)
+```
+
+## Variants
+
+### `minimumRequiredItems`
+
+This variant has the following behavior:
+
+- When `EditContainer` is visible, and the number of items in the array is one, the entier toolbar will be hidden.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx:77]:
```suggestion
- When `EditContainer` is visible, and the number of items in the array is one, the entire toolbar will be hidden.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx [hunk 12]
```diff
@@ -22,4 +24,74 @@ render(
...
+- When `ViewContainer` is visible, and the number of items in the array is one, the delete/remove button will be hidden.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx:78]:
```suggestion
- When `ViewContainer` is visible, and the number of items in the array is one, theremove button will be hidden.
```

I think it's safe to use remove button, as that's the name of the component as well `<Iterate.RemoveButton />`

File: packages/dnb-eufemia/src/extensions/forms/Iterate/Toolbar/Toolbar.tsx [hunk 13]
```diff
@@ -1,13 +1,85 @@
...
+  RemoveButton: typeof RemoveButton
+  CancelButton: typeof CancelButton
+  DoneButton: typeof DoneButton
+}
+export type Props = Omit<SpaceAllProps, 'children'> & {
+  children?: React.ReactNode | ((params: ToolbarParams) => React.ReactNode)
+  variant?: /**
+   * Use this variant to hide the toolbar when there is only one item in the array.
+   */
+  'minimumRequiredItems'
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/Toolbar/Toolbar.tsx:26]: Based on the descriptive line above, could this be a better more understandable prop name?

```suggestion
  'hidesWhenOneItem'
```

Personally, I still don't understand the name/word `minimumRequiredItems`.
I'll try to better understand this feature, and come up with a different name.

Maybe you can try to explain to me what happens when I as a user use the variant `minimumRequiredItems`?


```suggestion
  'theusecase'
```

By theusecase, I mean maybe the variant name should be "the use case", in which use case would I want to use this variant?

Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/Iterate/Toolbar/Toolbar.tsx]: Yes, lets find a meaningful variant or prop name.

The variant changes have been moved over to this PR #3919
This PR is only for the new container behavior and the fully customization of a iterate Toolbar.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx [hunk 14]
```diff
@@ -22,4 +24,46 @@ render(
 )
 ```
 
-The Toolbar is integrated into the containers [Iterate.ViewContainer](/uilib/extensions/forms/Iterate/ViewContainer/) and [Iterate.EditContainer](/uilib/extensions/forms/Iterate/EditContainer/).
+## Cusomize the Toolbar
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx:27]:
```suggestion
## Customize the Toolbar
```

**Review by langz — APPROVED** [review]
Thanks for all the great work with this PR, looks very good now 👏 🥇 

I do have a few thoughts(not necessarily directly related to this PR) that came about when testing the example/feature in the deploy preview again:

## Thought 1
What's the point of the "ferdig"-button is it just a "close button"? As it's not a "save button", since it doesn't only push/save the data when pressed. In the following video, I never have to press the "ferdig-buttons" for the data to be "stored/available":

https://github.com/user-attachments/assets/60368a10-5a89-4ccd-bd90-d1244719c8e5




## Thought 2
I get two different visual results if I in one scenario call this scenario 1:
- Click "legg til flere statsborgerskap" two times
- Select Finland in the 1st
- Danmark in the 2nd
- Angola in the 3rd. 

Vs. in scenario 2: 
- Select Finland in the 1st, and press "legg til flere statsborgerskap"
- Select Denmark in the 2nd and press "legg til flere statsborgerskap"
- Select Angola in the 3rd. 

In this screenshot scenario 1 is on the leftside and scenario 2 on the rightside:
Screenshot 2024-09-10 at 21 28 59

Is this intentional, or by design? I get that I've done two different sequence of actions, but the "result"(data-wise is the same {"countries":["FI","DK","AO"]}).

## Thought 3
Is it easy for an app to make it so that you can't add or select the same country twice, as you obviously can't be statsborger in the same country twice. This is just a thought 💭  


## Thought 4
Some times when I test with the keyboard only, using tab-between sections and arrow buttons for navigating down in the autocomplete, I experience that get sent to the "wrong element" after pressing enter on "Legg til flere statsborgerskap". In the following example below, the next active item that I get to when tabbing after pressing enter on "Legg til flere statsborgerskap" is "Endre", but I would expect that the new autocomplete(with label "velg et land") would be active:

https://github.com/user-attachments/assets/9a685856-bc96-4c9a-8171-3d9a916f7d01


## Thought 5
When tabbing through all elements, and pressing "Legg til flere statsborgerskap", sometimes the autocomplete field show a blank value/placeholder and sometimes it shows the velg et land placeholder. Not sure how/why it's different 🤔 
Seems like it only happens when I select a country in the first/initial field, then continue to click/press "Legg til flere statsborgerskap", but the logic of why some field get the label and other don't, I can't understand:
I think it perhaps could be related to the speed of which I was tabbing in? 🤔 


https://github.com/user-attachments/assets/d79b3430-720b-4bdf-bea8-32344c250e85




https://github.com/user-attachments/assets/c383f62b-e7d9-4474-b4fb-2dc2078d1901

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx [hunk 15]
```diff
@@ -21,6 +21,8 @@ import * as Examples from './Examples'
 
 ### Render props with primitive items
 
+You can provide the child as a function that receives the value of the item as the first argument, and the index of which item you are on as the second.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx:24]: And items as third?

Or maybe it doesn't add any value providing this info in the context of the example below?

**Comment by langz** [comment]
In my latest test now, https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open

I experience that the autocomplete/dropdown can take a few seconds to open upon clicking(wanting to open it):
Not sure what can cause this, and if it's only on my machine, but I would guess that my machine is way more powerful than most users of our components:


https://github.com/user-attachments/assets/3cd7daf0-1398-4344-98c3-5347c670aebd


https://github.com/user-attachments/assets/cbb395e8-67cf-42d3-a1d4-4a93583ec793

**Comment by langz** [comment]
When pressing cancel/avbryt or ferdig/done button after submitting a form with errors, should I as a user get some feedback? Or should I do as I do today, continue to see the same error message? It sort of feels strange pressing a button and nothing happens, there's like no feedback, even though the error message is already dispplayed, but could/should it be "re-displayed":

 

https://github.com/user-attachments/assets/52b58d72-012b-4eed-855d-ea130002a452

**Comment by tujoworker** [comment]
Very true! I added this back inn again ✅ Than you 🫶

**Comment by tujoworker** [comment]
> I now experience that the autocomplete/dropdown can take a few seconds to open upon clicking(wanting to open it)

I'm not able to reproduce it 🤔

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx (hunk 15 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/demos.mdx:24]: No, that looks ok. We did not make any changes to this one. I only think we need this info about the "function" as a child.

**Comment by tujoworker** [comment]
Can you test the keyboard usage again? Have made some improvements there.

**Comment by langz** [comment]
> > I now experience that the autocomplete/dropdown can take a few seconds to open upon clicking(wanting to open it)
> 
> 
> 
> I'm not able to reproduce it 🤔

Same here, maybe there was something going on here locally.

**Comment by langz** [comment]
https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open
Sometimes when I get the error message "Feilene ovenfor må rettes" it looks like it misses margin/spacing. But only in certain situations? In the following video, you can see it happen:


https://github.com/user-attachments/assets/678298a8-cf43-499d-8460-49358e228354

**Comment by langz** [comment]
> https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open

Unfortonately it did not seem to fix the problem.
In the following scenario, I would expect that the new autocomplete would become active, when tabbing after I've pressed enter on the "legg til flere statsborgerskap"-button:


In the following video I tab to and click enter at the "vis alternativer"-button before selecting a country.
image

https://github.com/user-attachments/assets/891f72c0-67d1-4058-8afa-a2ca8c667b76

In the following video, I just select country with arrow buttons after tabbed to it.
https://github.com/user-attachments/assets/66bf4337-2a91-41d6-b0d0-35f50367e1ef

**Comment by tujoworker** [comment]
> Unfortunately, it did not seem to fix the problem.

I'm not able to reproduce it. Lets have a chat about this.

**Comment by tujoworker** [comment]
ok, I did find the issue and made a fix ✅

Glad if you can test it again 🫶

File: packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/__tests__/PushContainer.test.tsx [hunk 16]
```diff
@@ -256,25 +304,35 @@ describe('PushContainer', () => {
       expect(buttons[1]).toHaveTextContent(
         nb.IterateEditContainer.cancelButton
       )
+
       // Hide the form
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/__tests__/PushContainer.test.tsx:308]:
```suggestion
      // Hide the form by adding a new item
```

File: packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/__tests__/PushContainer.test.tsx [hunk 17]
```diff
@@ -256,25 +304,35 @@ describe('PushContainer', () => {
...
+      )
+      expect(buttons).toHaveLength(2)
+      expect(buttons[0]).toHaveTextContent(
+        nb.IteratePushContainer.createButton
+      )
+      expect(buttons[1]).toHaveTextContent(
+        nb.IterateEditContainer.cancelButton
+      )
+
+      // Hide the form
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/Iterate/PushContainer/__tests__/PushContainer.test.tsx:331]:
```suggestion
      // Hide the form by adding a new item
```

**Comment by langz** [comment]
> https://eufemia-git-feat-iterate-container-mode-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#initially-open Sometimes when I get the error message "Feilene ovenfor må rettes" it looks like it misses margin/spacing. But only in certain situations? In the following video, you can see it happen:
> 
>  Screen.Recording.2024-09-11.at.23.06.03.mov 
> Looking like this(see the last error message): Screenshot 2024-09-12 at 09 32 01
> 
> I believe this is newly introduced, as I did not experience this last week.

I'm still able to reproduce this now.

**Pull request merged**